# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/workers/WorkerPrivate.cpp
# Commit: f3204b9ec0dc
# Full Hash: f3204b9ec0dc99aad004ce80499052fe5dcd5494
# Author: Yulia Startsev <ystartsev@mozilla.com>
# Date: 2023-01-19 04:46:07
# Regressor Bug: 1247687
# File Overlap Count: 1
# Description:
#   Bug 1247687 - Implement requestCredentials for Module Workers; r=jonco
#   
#   This weaves credentials through from the private to script creation.
#   
#   Depends on D155567
# ==============================================================================

diff -r 7327e6438343 -r f3204b9ec0dc dom/workers/WorkerPrivate.cpp
--- a/dom/workers/WorkerPrivate.cpp	Wed Jan 18 13:46:31 2023 +0000
+++ b/dom/workers/WorkerPrivate.cpp	Wed Jan 18 13:46:31 2023 +0000
@@ -2294,15 +2294,17 @@
 
 WorkerPrivate::WorkerPrivate(
     WorkerPrivate* aParent, const nsAString& aScriptURL, bool aIsChromeWorker,
-    WorkerKind aWorkerKind, enum WorkerType aWorkerType,
-    const nsAString& aWorkerName, const nsACString& aServiceWorkerScope,
-    WorkerLoadInfo& aLoadInfo, nsString&& aId, const nsID& aAgentClusterId,
+    WorkerKind aWorkerKind, RequestCredentials aRequestCredentials,
+    enum WorkerType aWorkerType, const nsAString& aWorkerName,
+    const nsACString& aServiceWorkerScope, WorkerLoadInfo& aLoadInfo,
+    nsString&& aId, const nsID& aAgentClusterId,
     const nsILoadInfo::CrossOriginOpenerPolicy aAgentClusterOpenerPolicy)
     : mMutex("WorkerPrivate Mutex"),
       mCondVar(mMutex, "WorkerPrivate CondVar"),
       mParent(aParent),
       mScriptURL(aScriptURL),
       mWorkerName(aWorkerName),
+      mCredentialsMode(aRequestCredentials),
       mWorkerType(aWorkerType),  // If the worker runs as a script or a module
       mWorkerKind(aWorkerKind),
       mLoadInfo(std::move(aLoadInfo)),
@@ -2545,16 +2547,18 @@
     const nsACString& aServiceWorkerScope, WorkerLoadInfo* aLoadInfo,
     ErrorResult& aRv, nsString aId) {
   return WorkerPrivate::Constructor(
-      aCx, aScriptURL, aIsChromeWorker, aWorkerKind, WorkerType::Classic,
-      aWorkerName, aServiceWorkerScope, aLoadInfo, aRv, std::move(aId));
+      aCx, aScriptURL, aIsChromeWorker, aWorkerKind, RequestCredentials::Omit,
+      WorkerType::Classic, aWorkerName, aServiceWorkerScope, aLoadInfo, aRv,
+      std::move(aId));
 }
 
 // static
 already_AddRefed<WorkerPrivate> WorkerPrivate::Constructor(
     JSContext* aCx, const nsAString& aScriptURL, bool aIsChromeWorker,
-    WorkerKind aWorkerKind, enum WorkerType aWorkerType,
-    const nsAString& aWorkerName, const nsACString& aServiceWorkerScope,
-    WorkerLoadInfo* aLoadInfo, ErrorResult& aRv, nsString aId) {
+    WorkerKind aWorkerKind, RequestCredentials aRequestCredentials,
+    enum WorkerType aWorkerType, const nsAString& aWorkerName,
+    const nsACString& aServiceWorkerScope, WorkerLoadInfo* aLoadInfo,
+    ErrorResult& aRv, nsString aId) {
   WorkerPrivate* parent =
       NS_IsMainThread() ? nullptr : GetCurrentThreadWorkerPrivate();
 
@@ -2616,8 +2620,8 @@
       ComputeAgentClusterIdAndCoop(parent, aWorkerKind, aLoadInfo);
 
   RefPtr<WorkerPrivate> worker = new WorkerPrivate(
-      parent, aScriptURL, aIsChromeWorker, aWorkerKind, aWorkerType,
-      aWorkerName, aServiceWorkerScope, *aLoadInfo, std::move(aId),
+      parent, aScriptURL, aIsChromeWorker, aWorkerKind, aRequestCredentials,
+      aWorkerType, aWorkerName, aServiceWorkerScope, *aLoadInfo, std::move(aId),
       idAndCoop.mId, idAndCoop.mCoop);
 
   // Gecko contexts always have an explicitly-set default locale (set by